### DISCUSSION TITLE
Decompile a delegate.

### DISCUSSION THREAD

**Discussion by cgiamouridisEncode** [icsharpcode/ILSpy#2581] [Development] [answered]
ILSpy can decompile assemblies. Can it decompile a delegate? I would imagine a CSharpDecompiler constructor overload accepting a Func or an Action as an argument instead of a "string fileName". 

Is it technically feasible, are there any future plans to support this? Any suggested work-arounds?

Thank you.

**Comment by siegfriedpammer** [comment, marked as answer]
It's not possible, because the decompiler requires a full PE image containing IL code and .NET metadata tables. What is your goal?

**Reply by cgiamouridisEncode** [reply]
Thank you. I was imagining something like:

```
Action<int> myAction = i =>
{
// body to be decompiled
};
var decompiler = new CSharpDecompiler(myAction, new DecompilerSettings());
decompiler.DecompileAsString();
```

without the need to specify its containing assembly.

Since, this is not technically feasible, I suspect even logically incompatible, I will follow a different strategy. I will use member methods instead of delegates and decompile them referring their own assembly.

Still, if myAction above is a local variable or even a method parameter, is it possible to dig in the assembly and get its MetadataToken to decompile it?

Thank you.
